### ISSUE TITLE
static webpage generation

### ISSUE THREAD

**Issue by markus2330** [ElektraInitiative/libelektra#3470]
@Namoshek it looks like there are some static webpage generators for Angular JS. Do you think it is feasible to use them for our setup?

Goal would be that people are able to view our webpage without JS.

**Comment by Namoshek** [comment]
A static web page doesn't really imply that there cannot be any scripts on a website. The concept just avoids generating websites dynamically, e.g. from database contents. And that is nothing we are currently using, as far as I am aware of. So the site actually is somewhat static already.

If you want to serve the website without any scripts, you can probably run a generator over the repository which creates HTML files from the markdown files and links them together. Maybe you could even use part of the existing preparation scripts (e.g. making paths in the markdown files absolute). But that is nothing I have experience with or what I would support, to be honest. It increases the required maintenance effort and if someone really wants to read the plain documentation, they can still jump over to GitHub. It's not script free either, but at least well-known and probably quite trusted...

**Comment by markus2330** [comment]
Yes, creating a new way of rendering Markdown is not what I want, we are very happy with the website as it is. (Except of some content improvements hopefully soon done by @d3nwp.)

The idea would be to run the SPA scripts locally for every page and store these pre-rendered pages. So basically "doing what the browser does", but during the build process. Of course this makes the build process one step longer...

If this is not feasible, let us simply close the issue. I thought that people might come up with such a solution, as our Angular JS version is probably already out of lifetime.
